runtime/permissions/mod.rs

```diff
@@ -226,22 +228,263 @@ impl AsRef<str> for EnvVarName {
   }
 }
 
+pub trait Descriptor: Eq + Clone {
+  fn flag_name() -> &'static str;
+  fn name(&self) -> Cow<str>;
+  fn stronger_than(&self, other: &Self) -> bool {
+    self == other
+  }
+}
+
 #[derive(Clone, Debug, Eq, PartialEq)]
-pub struct UnaryPermission<T: Eq + Hash> {
-  pub name: &'static str,
-  pub description: &'static str,
-  pub global_state: PermissionState,
+pub struct UnaryPermission<T: Descriptor + Hash> {
+  pub granted_global: bool,
   pub granted_list: HashSet<T>,
+  pub denied_global: bool,
   pub denied_list: HashSet<T>,
+  pub no_to_prompt_global: bool,
+  pub no_to_prompt_list: HashSet<T>,
   pub prompt: bool,
 }
 
+impl<T: Descriptor + Hash> Default for UnaryPermission<T> {
+  fn default() -> Self {
+    UnaryPermission {
+      granted_global: Default::default(),
+      granted_list: Default::default(),
+      denied_global: Default::default(),
+      denied_list: Default::default(),
+      no_to_prompt_global: Default::default(),
+      no_to_prompt_list: Default::default(),
```

Comment:
We should rework these names since 'denied' and 'no to prompt' aren't really different.

How about `denied` -> `flag_denied` and `no_to_prompt` -> `request_denied`?

Reply:
Good idea. How about `no_to_prompt` -> `prompt_denied`?

Reply:
Yeah that's better 👍